include/gridtools/fn/backend/gpu.hpp

```diff
@@ -149,19 +149,22 @@ namespace gridtools::fn::backend {
             auto strides = sid::get_strides(std::forward<Composite>(composite));
 
```

Comment:
Maybe something like the following (as the first thing in the function) would be more readable. What do you think?
```suggestion

if empty_domain(sizes)
    return;

```

Reply:
and maybe even better to move it outside of gpu.hpp to the common part where we are dispatching to the backends.

Reply:
I created an `is_domain_empty` function. I usually use an `if` block instead of an early return when the code is simple, but I'm happy with either.

Reply:
I prefer Hannes early return in this case as I my eyes don’t have to track any braces opening and closing. And my proposed changes would be even a bit more minimalistic, just before the kernel launch:
```cpp
if (blocks.x ==0 || blocks.y == 0 || blocks.z == 0)
    return;
```
(of course this does not include the optimization of skipping unnecessary computation of ptr_holder, strides, etc., but I don’t think we need any optimization of such trivial computations for an empty domain)

Reply:
Well, uhh, it was with the `blocks` before Hannes suggested to make it a function :) Now that I wrote the empty domain function, I'd rather keep it, because that satisfies the requirements independent of how you calculate the blocks.

Reply:
I'll also put an early return